### PR TITLE
chore: move tests in reth_execution_types::chain to reth-evm-optimism

### LINKED ISSUE #11111: Copy tests in `reth_execution_types::chain` to `reth-evm-optimism`
### Describe the feature

Copy tests with optimism feature in `reth_execution_types::chain` into `reth-evm-optimism`. Remove the optimism feature gated code form the original tests (and remove code feature gated by `#[cfg(not(feature = "optimism"))]` in the `reth-evm-optimism` copy).

This will likely cause issues in CI, for now the original tests will have to be feature gated with `#[cfg(not(feature = "optimism"))]`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by 0xriazaka**
Closes https://github.com/paradigmxyz/reth/issues/11111

### CODE REVIEW

File: crates/optimism/evm/src/lib.rs [hunk 1]
```diff
@@ -545,4 +546,76 @@ mod tests {
...
+        // Clone the default block into block1 and block2
+        let mut block1 = block.clone();
+        let mut block2 = block;
+
+        // Set the hashes of block1 and block2
+        block1.block.header.set_hash(block1_hash);
+        block2.block.header.set_hash(block2_hash);
+
+        // Create a random receipt object, receipt1
+        let receipt1 = Receipt {
```
Comment (onbjerg) [inline crates/optimism/evm/src/lib.rs:568]: this is missing the deposit nonce and deposit receipt version fields which are unique to optimism

**Review by onbjerg — DISMISSED** [review]

**Review by mattsse — CHANGES_REQUESTED** [review]
ty, this is helpful, the only thing we need to do is as pointed out, also port the additional optimism receipt fields

**Review by mattsse — APPROVED** [review]

**Review by mattsse — CHANGES_REQUESTED** [review]
this does not compile yet, needs same changes, like private API on `Chain`

**Comment by 0xriazaka** [comment]
> this does not compile yet, needs same changes, like private API on `Chain`

wdm with "same changes"

**Review by emhane — APPROVED** [review]

**Comment by emhane** [comment]
how's it going with this, need any help?

**Review by mattsse — APPROVED** [review]

**Pull request merged**